weather_dl/download_pipeline/partition.py

```diff
@@ -141,14 +142,14 @@ def prepare_partitions(config: Config) -> t.Iterator[Config]:
         yield _create_partition_config(option, config)
 
 
-def new_downloads_only(candidate: Config, store: t.Optional[Store] = None) -> bool:
+def new_downloads_only(candidate: Config, store: t.Optional[Store] = None) -> t.Iterator[Config]:
     """Predicate function to skip already downloaded partitions."""
     if store is None:
         store = FSStore()
     should_skip = skip_partition(candidate, store)
```

Comment:
Thinking about this some more, I don't think my idea is a good approach. This is because we use this function later on to prevent double download. Since the state changes at that time, listing and caching will produce the wrong behavior.

Reply:
Since the thread approach may not be as fruitful as we originally expected, what if we try the following: We create two methods to check for skipping – `skip_partition` and `bulk_skip`. The latter function can take the file listing approach I mention above, and will only be used for the partition stage of the pipeline. Then, we can keep the existing method for its use downstream in the pipeline (double checking work during fetches).